activation/handler.go

```diff
@@ -569,9 +557,12 @@ func (h *Handler) handleAtx(
 		return nil, fmt.Errorf("%w: %w", errMalformedData, err)
 	}
 
-	atx := types.ActivationTxFromWireV1(&atxOnWire)
+	atx, err := types.ActivationTxFromWireV1(&atxOnWire)
+	if err != nil {
+		return nil, fmt.Errorf("%w: %w", errMalformedData, err)
+	}
```

Comment:
This conversion should happen after the ATX was validated and before it is stored in the DB. Validation for ATXv1 will differ from validation for ATXv2 - which also means we will have different methods for validation based on the version of ATX that is validated.

This don't need to be a `handleV1` and `handleV2` method, but this could just be one service per wiretype version that we inject into the handler that handles ATXs of the specified version.

Reply:
Yes, I agree, but this PR is focused on refactoring/simplifying `types.ActivationTx`, not overhauling everything. I'm trying to divide the gigantic work into separate, reviewable steps (PRs) that we could iterate ~fast, yet they are already big and slow to review apparently.

Reply:
I would do it in smaller steps first:

1. Create the ATXv1 wire type and only use it in the `activation.Handler`
2. Update `sql/atx` functions to not require scale encoding/decoding any more
  2a. This requires a DB migration that extracts info not currently present in the DB from the encoded ATXs
  2b. This can also be used to separate the atx blobs from the rest of the data into their own table
3. Update `types.ActivationTx` to not include NIPost and other unused fields any more

Reply:
EDIT updated comment with more info